### ISSUE TITLE
How to set random seed

### ISSUE THREAD

**Issue by shiratsu** [tflearn/tflearn#757]
I would like to know how to set random seed in tflearn.
Because,in current code,my program produce different result each time.
for example,chainer,i can get a same result by setting random seed.
```
random.seed(1)
np.random.seed(1)
```

I guess that the following code might be effective.
However,the result is still differnt each time.

```
tf.set_random_seed(1)
tflearn.init_graph(seed=1)
```

How do i solve this problem?

**Comment by emilwallner** [comment]
I have the same problem...

**Comment by emilwallner** [comment]
```
from __future__ import division, print_function, absolute_import

import tflearn
import tensorflow as tf
import numpy as np
from tflearn.layers.core import input_data, dropout, fully_connected
from tflearn.layers.conv import conv_2d, max_pool_2d
from tflearn.layers.normalization import local_response_normalization
from tflearn.layers.estimator import regression
tflearn.init_graph(seed=1)
tf.set_random_seed(1)
np.random.seed(1)
tf.reset_default_graph()

# Data loading and preprocessing
import tflearn.datasets.mnist as mnist
X, Y, testX, testY = mnist.load_data(one_hot=True)
X = X.reshape([-1, 28, 28, 1])
testX = testX.reshape([-1, 28, 28, 1])

# Building convolutional network
network = input_data(shape=[None, 28, 28, 1], name='input')
network = conv_2d(network, 32, 3, activation='relu', regularizer="L2")
network = max_pool_2d(network, 2)
network = local_response_normalization(network)
network = conv_2d(network, 64, 3, activation='relu', regularizer="L2")
network = max_pool_2d(network, 2)
network = local_response_normalization(network)
network = fully_connected(network, 128, activation='tanh')
network = dropout(network, 0.8)
network = fully_connected(network, 256, activation='tanh')
network = dropout(network, 0.8)
network = fully_connected(network, 10, activation='softmax')
network = regression(network, optimizer='adam', learning_rate=0.01,
                     loss='categorical_crossentropy', name='target')

# Training
model = tflearn.DNN(network, tensorboard_dir='/output', tensorboard_verbose=0)
model.fit({'input': X}, {'target': Y}, n_epoch=1,
           validation_set=({'input': testX}, {'target': testY}),
           snapshot_step=100, show_metric=True, run_id='test_2_seed_2')
```
![screen shot 2017-08-08 at 3 32 10 pm](https://user-images.githubusercontent.com/12543699/29074249-d554fa9a-7c4e-11e7-8d02-5925d8fb4ef9.png)

**Comment by emilwallner** [comment]
```
from __future__ import division, print_function, absolute_import

import tflearn
import tensorflow as tf
import numpy as np
from tflearn.layers.core import input_data, dropout, fully_connected
from tflearn.layers.conv import conv_2d, max_pool_2d
from tflearn.layers.normalization import local_response_normalization
from tflearn.layers.estimator import regression
tf.reset_default_graph()
tflearn.init_graph(seed=1)
tf.set_random_seed(1)
np.random.seed(1)


# Data loading and preprocessing
import tflearn.datasets.mnist as mnist
X, Y, testX, testY = mnist.load_data(one_hot=True)
X = X.reshape([-1, 28, 28, 1])
testX = testX.reshape([-1, 28, 28, 1])

# Building convolutional network
network = input_data(shape=[None, 28, 28, 1], name='input')
network = conv_2d(network, 32, 3, activation='relu', regularizer="L2")
network = max_pool_2d(network, 2)
network = local_response_normalization(network)
network = conv_2d(network, 64, 3, activation='relu', regularizer="L2")
network = max_pool_2d(network, 2)
network = local_response_normalization(network)
network = fully_connected(network, 128, activation='tanh')
network = dropout(network, 0.8)
network = fully_connected(network, 256, activation='tanh')
network = dropout(network, 0.8)
network = fully_connected(network, 10, activation='softmax')
network = regression(network, optimizer='adam', learning_rate=0.01,
                     loss='categorical_crossentropy', name='target')

# Training
model = tflearn.DNN(network, tensorboard_dir='/output', tensorboard_verbose=0)
model.fit({'input': X}, {'target': Y}, n_epoch=1,
           validation_set=({'input': testX}, {'target': testY}),
           snapshot_step=100, show_metric=True, run_id='test_3_seed_2')
```
![screen shot 2017-08-08 at 3 34 43 pm](https://user-images.githubusercontent.com/12543699/29074336-3233cbce-7c4f-11e7-918a-956977805894.png)

**Comment by SaadSen** [comment]
Have you found any solutions yet? I'm having same issues.

**Comment by emilwallner** [comment]
@SaadSen I noticed that it worked on my CPU, but not when [I ran it on a cloud GPU](https://github.com/tflearn/tflearn/issues/865). I believe it has something to do with how calculations are made on a GPU. Ping me if you figure it out.

**Comment by SaadSen** [comment]
I see small variations even on my CPU. Is it suppose to be like this? 

<img width="907" alt="screen shot 2017-09-27 at 7 07 02 pm" src="https://user-images.githubusercontent.com/28780598/30927039-6a53eec8-a3b7-11e7-9842-1afa20a342dc.png">

**Comment by emilwallner** [comment]
Did you fix the seed and  'shuffle=False' ? I was able to have two exact runs with those settings. I also heard that setting the seed before importing the libraries have an effect, but I haven't validated it.

**Comment by DexHunter** [comment]
The random seed doesn't work for me as well. I tried `np.random.seed(0)` and `tf.set_random_seed(0)`. I wonder if there are any updates on this issue? Thanks!
